src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -1123,6 +1123,12 @@ def to_snowpark(
             index, index_label
         )
 
+    def cache_result(self) -> None:
```

Comment:
should this return not the SnowflakeQueryCompiler itself/a cached version of it? Can we add some comments about temp table and that this lives for the duration of the session?